projects/core/src/lib/layer-collections/layer-collection.service.ts

```diff
@@ -45,4 +76,186 @@ export class LayerCollectionService {
+        this.resolveLayer(layerId).subscribe((layer: Layer) => {
+            this.projectService.addLayer(layer);
+        });
+    }
+
+    private resolveLayer(layerId: ProviderLayerIdDict): Observable<Layer> {
+        return this.getLayer(layerId.providerId, layerId.layerId).pipe(
+            mergeMap((layer: LayerDict) =>
+                combineLatest([of(layer), this.registerAndGetLayerWorkflowId(layerId.providerId, layerId.layerId)]),
+            ),
+            mergeMap(([layer, workflowId]: [LayerDict, UUID]) =>
+                combineLatest([of(layer), of(workflowId), this.projectService.getWorkflowMetaData(workflowId)]),
+            ),
+            map(([layer, workflowId, resultDescriptorDict]: [LayerDict, UUID, ResultDescriptorDict], _i) => {
+                const keys = Object.keys(resultDescriptorDict);
+                if (keys.includes('columns')) {
+                    return new VectorLayer({
+                        name: layer.name,
+                        workflowId,
+                        isVisible: true,
+                        isLegendVisible: false,
+                        symbology: layer.symbology
+                            ? VectorSymbology.fromVectorSymbologyDict(layer.symbology as VectorSymbologyDict)
+                            : this.createVectorSymbology((resultDescriptorDict as VectorResultDescriptorDict).dataType),
+                    });
+                } else if (keys.includes('measurement')) {
+                    return new RasterLayer({
+                        name: layer.name,
+                        workflowId,
+                        isVisible: true,
+                        isLegendVisible: false,
+                        symbology: layer.symbology
+                            ? RasterSymbology.fromRasterSymbologyDict(layer.symbology as RasterSymbologyDict)
+                            : RasterSymbology.fromRasterSymbologyDict({
+                                  type: 'raster',
+                                  opacity: 1.0,
+                                  colorizer: {
+                                      type: 'linearGradient',
+                                      breakpoints: [
+                                          {value: 1, color: [0, 0, 0, 255]},
+                                          {value: 255, color: [255, 255, 255, 255]},
+                                      ],
+                                      defaultColor: [0, 0, 0, 0],
+                                      noDataColor: [0, 0, 0, 0],
+                                  },
+                              }),
+                    });
+                } else {
+                    // TODO: implement plots, etc.
+                    throw new Error('Adding this workflow type is unimplemented, yet');
+                }
+            }),
+        );
+    }
+
+    private doAddLayerToProject(layer: LayerDict, workflowId: UUID, resultDescriptorDict: ResultDescriptorDict): void {
```

Comment:
When is `addLayerToProject` and `doAddLayerToProject` used?

Reply:
this is a private helper method, used by addLayerToProject which is public

Reply:
But why do both exist here? It is not used by addLayerToProject.

It looks like two methods do the same thing slightly different.

Reply:
you are right they were redundant e3d9720bde4caf0444077f523258e65bd9b984dc